weldx/asdf/util.py

```diff
@@ -141,30 +141,38 @@ def write_read_buffer(
     return read_buffer(buffer, open_kwargs)
 
 
-def get_yaml_header(file) -> str:
+def get_yaml_header(file, parse=False) -> Union[str, dict]:
     """Read the YAML header part (excluding binary sections) of an ASDF file.
 
     Parameters
     ----------
-    file
+    file :
         filename, ``pathlib.Path`` or ``BytesIO`` buffer of ASDF file
 
+    parse :
+        if True, returns the interpreted YAML header as dict
     Returns
     -------
-    str
-        The YAML header the ASDF file
+    str, dict
+        The YAML header as string the ASDF file, if parse is False. Or if parse is True,
+        return the parsed header.
 
     """
+
+    def read_header(handle):
+        # reads lines until the byte string "...\n" is approached.
+        return b"".join(iter(handle.readline, b"...\n"))
+
     if isinstance(file, BytesIO):
         file.seek(0)
-        code = file.read()
+        code = read_header(file)
     else:
         with open(file, "rb") as f:
-            code = f.read()
+            code = read_header(f)
 
-    parts = code.partition(b"\n...")
-    code = parts[0].decode("utf-8") + parts[1].decode("utf-8")
-    return code
+    if parse:
+        return yaml.load(code, yaml.BaseLoader)
```

Comment:
```suggestion
        return asdf.yamlutil.load_tree(code)
```
I have been wondering about this for a while, turns out it is quite simple

With the change above we will return a `asdf.tagged.TaggedDict` which is basically a dict that keeps all the ASDF `tag` definitions in a `_tag` attribute (for all nested types)
This allows us to keep all the tag/class information from the yaml header as well 🚀

Reply:
That's really nice, cheers!

Reply:
we should keep this in mind when handling the extracted YAML sections from files in KISA, this should give us all the original file information needed without the binary bits @marscher

Reply:
Yes. I changed this function precisely for the broad usage in KISA. When we will be handling lots of large files it would have been overkill to read them all to memory just for the sake of extracting the small header.